docs/troubleshooting.md

```diff
@@ -0,0 +1,66 @@
+# Troubleshooting
+
+## Enable Debug Logging
+
+When things are not working, a good first step is to restart the program with
+debug logging enabled. Do this by setting the `OTEL_LOG_LEVEL` environment
+variable to `debug`.
+
+```sh
+export OTEL_LOG_LEVEL="debug"
+```
+
+Make sure to unset this logging level after the issue is resolved, it is quite
+verbose and likely will overload systems if left on indefinitely.
+
+## Missing Spans
+
+There are many points were spans may be dropped. Follow these steps to ensure
+they are not being dropped by the `SDK`.
+
+1. [Enable debug logging](#enable-debug-logging). This should log the `SDK`
+   configuration and all spans sent to registered `SpanProcessor`s. Verify the
+   correct number of span are reported as exported in the log messages.
+   Additionally, pay attention to any `total_dropped` count.
+2. If an incorrect number of spans are being reported as exported, verify you
+   are ending the spans you expect to see.
+3. If spans are reported as dropped, the `BatchSpanProcessor` is dropping
```

Comment:
```suggestion
3. If spans are reported as dropped, the `BatchSpanProcessor` is dropping
```
This seems to be the key step. Is counting exported spans necessary (steps 1 and 2) or just checking whether they are being exported in general or not?

I guess that from this procedure one might get two outcomes:

- If all spans are dropped, it might be due to a configuration issue.
- If only some spans are being dropped, you might need to increase the span processor capacity. For this, it'd help to get concrete instructions with examples (what env vars would you modify and how?)

Reply:
There is another situation this is trying to address: no spans are exported or dropped. This can be the case when users are not correctly ending a span (or the SDK sampling is misconfigured which shouldn't happen given we do not expose this).

I can take another pass at this to give more context.

Reply:
That'd be great if you'd do it, @MrAlias ! I think it'd be helpful to split it into three subheadings, one per situation.